src/Components/Login/Login.tsx

```diff
@@ -19,56 +13,61 @@ const ALL_USERS = gql `
   }
 `
 
-const ALL_BOOKS = gql `
-  query AllBooks {
-    books {
-      id
-      imageUrl
-    }
-  }
-`
-
 const Login = () => {
   const [email, setEmail] = useState('')
   const [username, setUsername] = useState('')
   const [userLocation, setUserLocation] = useState('')
   const [activeAccount, setAccount] = useState(true)
-  const [allUsers, setUsers] = useState([{userName: "Josh", emailAddress: "test", location: "Denver"}])
+  const [loginError, setLoginError] = useState(false)
+  const [errorMessage, setErrorMessage] = useState('')
+  const [allUsers, setUsers] = useState([{emailAddress: "joshua@gmail.com", userName: "Joshua", location: "Colorado"}])
 
   let navigate = useNavigate()
 
-  // const { data, loading, error } = useQuery(ALL_USERS)
-  const { data, loading, error } = useQuery(ALL_BOOKS)
-
-  console.log("Endpoint", error)
+  const { data, loading, error } = useQuery(ALL_USERS)
   
-
   const verifyLogin = () => {
      let noUser = allUsers.map(user => {
```

Comment:
It looks like this wasn't something added by your PR, but I think it's worth pointing out.

The `.map()` function takes an array, transforms it, and gives you another array.
For example, if I have an array of numbers, and I want another array of double those specific numbers...
```TS
const numbers = [1, 5, 15];
const doubledNumbers = numbers.map((num) => num * 2);

console.log(numbers);
// [1, 5, 15]
console.log(doubledNumbers);
// [2, 10, 30]
```

The reason I bring it up is because this implementation isn't returning anything. Instead it looks like you guys are looping through the data, and determining if the inputed email is within the user array. Here's what we can use instead:

The `.some()` function on arrays allows us to check the entirety of the array for a condition, and if "some" (at least one) of the array items meets true on that condition, the function will return true. So we can refactor this logic to be:

```TS
if (allUsers.some((user) => user.emailAddress === email) {
  // We got a match, log them in!
  setAccount(true);
  navigate("/home");
} else {
  // Not a single email matched, show an error
  setErrorMessage("Could not find that email. Please try again.");
}
```

Please let me know if you have questions on this

Reply:
Thank you, a lot of this logic is actually going to change. The backend will be handling some of this information. I will be sending the user email that is attempting to sign in and the backend will search for that email and return either a 200 status or an error. I will then proceed based on that return.